modules/bitcoin/module.cpp

```diff
@@ -366,5 +368,88 @@ std::optional<std::string> Bitcoin::addrv2_parse(std::span<const uint8_t> buffer
            "cjdns=" + std::to_string(cjdns) + "i2p=" + std::to_string(i2p);
 }
 
+std::optional<std::string> Bitcoin::psbt_parse(std::span<const uint8_t> buffer) const {
+
+    if (buffer.empty()) {
+        return std::nullopt;
+    }
+
+    PartiallySignedTransaction psbt;
+    std::string error;
+
+    // Attempt to decode raw psbt from buffer
+    if(!DecodeRawPSBT(psbt, MakeByteSpan(buffer), error)) {
+        return std::nullopt;
```

Comment:
If Bitcoin Core could not decode a PSBT or the transaction is not valid we should not return a nullopt. It should return something that indicates that it fails (e.g. an empty string). We only use `std::nullopt` to sinalize that a module does not support something.

Reply:
Correct, I'll fix that!

Reply:
Done!